### PR TITLE
Adding accessibility support for commands.dev

### PR DESCRIPTION
**Pull request by ianhodge**
Based on this audit by Alex: https://docs.google.com/document/d/19hXOfvJr1VuWQdRsecguZYbTG2lGHyGH/edit
Note: The suggestion of changing to "Contribute on Github" does not fit on mobile, might need to brainstorm another solution here
Fixes: WAR-3488

### CODE REVIEW

File: components/navbar.tsx [hunk 1]
```diff
@@ -66,36 +64,38 @@ export default function NavBar() {
...
+                      action: "click_on_navbar_github",
+                      category: "Click on GitHub",
+                      label: "Click on GitHub via NavBar",
+                      value: window.location.pathname,
+                    });
+                  }}
+                >
                   <span className="text-l font-normal mt-4 text-back dark:text-white">
-                    About
+                    Contribute
```
Comment (alokedesai) [inline components/navbar.tsx:90]: I think per Alex's feedback this should be "Contribute on GitHub"--see item 3) in his doc

File: components/search/MobileSearchBox.tsx [hunk 2]
```diff
@@ -31,6 +31,7 @@ function SearchBox({
       <input
         id={"algolia_search"}
         type="search"
+        aria-label="Search for commands"
```
Comment (alokedesai) [inline components/search/MobileSearchBox.tsx:34]: just confirming this is all we need to do here? He had feedback around using a "for" attribute and making it clear that enter submits the form. Not sure if the "for" attribute is useful here though:

> There is a search form (I believe) that reads as Ctrl+K. This form should have a valid label saying something like “Search for Commands”. The label should be associated with the input field via “for” attribute. The form should have a “Search Commands” submit button located below the input field as Enter key should not be implied.

File: components/navbar.tsx [hunk 3]
```diff
@@ -1,18 +1,16 @@
 import Link from "next/link";
 import React, { useState } from "react";
 import DarkModeToggle from "./DarkModeToggle";
```
[the remark this answers is not in the document]
Comment (ianhodge) [reply components/navbar.tsx:3]: ahhhhh - that's what he was referencing. I can add an aria-label and description to it

File: components/navbar.tsx (hunk 1 above)
Comment (ianhodge) [reply components/navbar.tsx:90]: I mentioned this in the PR description, but "Contribute on github" does not fit on mobile, we might have to come up with some other content here

File: components/search/MobileSearchBox.tsx (hunk 2 above)
Comment (ianhodge) [reply components/search/MobileSearchBox.tsx:35]: just did a quick google - misunderstood the ask on first pass. Should be addressed now

**Review by alokedesai — APPROVED** [review]

**Pull request merged**